torch/csrc/Exceptions.h

```diff
@@ -69,6 +69,23 @@ struct python_error : public std::exception {
     }
   }
 
+  virtual const char* what() const noexcept override {
+    // Retrieve the error message from the value.
+    if (value != nullptr && value != Py_None) {
+      AutoGIL gil;
+      Py_XINCREF(value);
+      PyObject* pyStr = PyObject_Str(value);
+      if (pyStr != Py_None && pyStr != nullptr) {
+        PyObject* str = PyUnicode_AsEncodedString(pyStr, "utf-8", "strict");
+        const auto& ret = PyBytes_AS_STRING(str);
```

Comment:
You can't do this. You need to keep `str` live for the returned `char*` to be valid, but there's no way in this API to do so. Your `char*` goes immediately to dead data after you do this